server/sonar-server/src/main/java/org/sonar/server/organization/ws/UpdateProjectVisibilityAction.java

```diff
@@ -74,9 +78,18 @@ public void handle(Request request, Response response) throws Exception {
       Optional<OrganizationDto> optionalOrganization = dbClient.organizationDao().selectByKey(dbSession, organizationKey);
       OrganizationDto organization = checkFoundWithOptional(optionalOrganization, "No organization with key '" + organizationKey + "' can be found.");
       userSession.checkPermission(OrganizationPermission.ADMINISTER, organization.getUuid());
+      checkCanUpdateProjectsVisibility(organization, newProjectsPrivate);
       dbClient.organizationDao().setNewProjectPrivate(dbSession, organization, newProjectsPrivate);
       dbSession.commit();
     }
     response.noContent();
   }
+
+  private void checkCanUpdateProjectsVisibility(OrganizationDto organization, boolean newProjectsPrivate) {
+    try {
+      billingValidations.checkCanUpdateProjectVisibility(new BillingValidations.Organization(organization.getKey(), organization.getUuid()), newProjectsPrivate);
+    } catch (BillingValidations.BillingValidationsException e) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Either log or rethrow this exception. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS1166)